python/lancedb/query.py

```diff
@@ -185,14 +185,33 @@ def to_df(self) -> "pd.DataFrame":
         """
         return self.to_pandas()
 
-    def to_pandas(self) -> "pd.DataFrame":
+    def to_pandas(self, flatten: Optional[Union[int, bool]] = None) -> "pd.DataFrame":
         """
         Execute the query and return the results as a pandas DataFrame.
         In addition to the selected columns, LanceDB also returns a vector
         and also the "_distance" column which is the distance between the query
         vector and the returned vector.
+
+        Parameters
+        ----------
+        flatten: Optional[Union[int, bool]]
+            If flatten is True, flatten all nested columns.
+            If flatten is an integer, flatten the nested columns up to the
+            specified depth.
+            If unspecified, do not flatten the nested columns.
         """
-        return self.to_arrow().to_pandas()
+        tbl = self.to_arrow()
+        if (flatten == -1) or (flatten is True):
```

Comment:
Minor nit: maybe `flatten < 0` instead of `flatten == -1`? (or just return an error if flatten is <= 0?, maybe suggest setting it to `True`)

Reply:
sg, i'll just raise ValueError if flatten is <= 0